pkg/clientgen/testdata/tsapp/expected_shared.ts

```diff
@@ -0,0 +1,852 @@
+// Code generated by the Encore v0.0.0-develop client generator. DO NOT EDIT.
+
+// Disable eslint, jshint, and jslint for this file.
+/* eslint-disable */
+/* jshint ignore:start */
+/*jslint-disable*/
+
+/**
+ * BaseURL is the base URL for calling the Encore application's API.
+ */
+export type BaseURL = string
+
+export const Local: BaseURL = "http://localhost:4000"
+
+/**
+ * Environment returns a BaseURL for calling the cloud environment with the given name.
+ */
+export function Environment(name: string): BaseURL {
+    return `https://${name}-app.encr.app`
+}
+
+/**
+ * PreviewEnv returns a BaseURL for calling the preview environment with the given PR number.
+ */
+export function PreviewEnv(pr: number | string): BaseURL {
+    return Environment(`pr${pr}`)
+}
+
+/**
+ * Client is an API client for the app Encore application.
+ */
+export default class Client {
+    public readonly svc: svc.ServiceClient
+
+
+    /**
+     * Creates a Client for calling the public and authenticated APIs of your Encore application.
+     *
+     * @param target  The target which the client should be configured to use. See Local and Environment for options.
+     * @param options Options for the client
+     */
+    constructor(target: BaseURL, options?: ClientOptions) {
+        const base = new BaseClient(target, options ?? {})
+        this.svc = new svc.ServiceClient(base)
+    }
+}
+
+/**
+ * Import the auth handler to be able to derive the auth type
+ */
+import type { auth as auth_Auth } from "~backend/svc/svc";
+
+/**
+ * ClientOptions allows you to override any default behaviour within the generated Encore client.
+ */
+export interface ClientOptions {
+    /**
+     * By default the client will use the inbuilt fetch function for making the API requests.
+     * however you can override it with your own implementation here if you want to run custom
+     * code on each API request made or response received.
+     */
+    fetcher?: Fetcher
+
+    /** Default RequestInit to be used for the client */
+    requestInit?: Omit<RequestInit, "headers"> & { headers?: Record<string, string> }
+
+    /**
+     * Allows you to set the authentication data to be used for each
+     * request either by passing in a static object or by passing in
+     * a function which returns a new object for each request.
+     */
+    auth?: RequestType<typeof auth_Auth> | AuthDataGenerator
+}
+
+/**
+ * Import the endpoint handlers to derive the types for the client.
+ */
+import {
+    dummy as api_Svc_Svc_Dummy,
```

Comment:
Why are all the names CamelCased? Doesn't look very idiomatic. Or is that just the input endpoint name?

Reply:
escape bug. fixed